tests/manual/util/src/com/oracle/util/testing/ManualTestWindow.java

```diff
@@ -0,0 +1,279 @@
+        public Builder runAfter(Runnable r) {
+            ManualTestWindow.runAfter = r;
+            return this;
+        }
+
+        /**
+         * Creates and launches the test application and its main window.
+         * This method throws a {@link NullPointerException} if either instructions
+         * or test node generator is {@code null}.
+         */
+        public void buildAndRun() {
+            Objects.requireNonNull(instructions, "instructions must not be null");
+            Objects.requireNonNull(generator, "generator must not be null");
+            Application.launch(TApplication.class);
+        }
+    }
+
+    /** The Application to run the manual test */
+    public static class TApplication extends Application {
+        public TApplication() {
+        }
+
+        @Override
+        public void start(Stage stage) throws Exception {
+            Node content = generator.get();
+
+            BlurType blurType = BlurType.GAUSSIAN;
+            Color color = Color.gray(0, 0.5);
+            double radius = 10;
+            double spread = 0;
+            double offsetX = 1;
+            double offsetY = 1;
+            DropShadow shadow = new DropShadow(blurType, color, radius, spread, offsetX, offsetY);
+
+            BorderPane cp = new BorderPane(content);
+            cp.setMargin(content, new Insets(10));
+            cp.setBackground(Background.fill(Color.gray(1)));
+            cp.setEffect(shadow);
+
+            Node instructionField = toTextFlow(instructions);
+
+            Region fill = new Region();
+
+            Button screenshotButton = new Button("Screenshot");
+            screenshotButton.setMinWidth(100);
+            screenshotButton.setOnAction((ev) -> {
+                // TODO
+            });
+            screenshotButton.setDisable(true); // TODO
+
+            Button failButton = new Button("✘ Fail");
+            failButton.setMinWidth(100);
+            failButton.setOnAction((ev) -> {
+                // TODO encoded screenshot to stderr?
+                Platform.exit();
+                throw new AssertionError("Failed Manual Test: " + title);
+            });
+
+            Button passButton = new Button("✔ Pass");
+            passButton.setMinWidth(100);
+            passButton.setOnAction((ev) -> {
+                Platform.exit();
+            });
+
+            HBox buttons = new HBox(
+                10,
+                screenshotButton,
+                fill,
+                failButton,
+                passButton
+            );
+            HBox.setHgrow(fill, Priority.ALWAYS);
+
+            VBox vb = new VBox(
+                10,
+                instructionField,
+                cp,
+                buttons
+            );
+            vb.setPadding(new Insets(10));
+            VBox.setVgrow(cp, Priority.ALWAYS);
+
+            Scene scene = new Scene(vb);
+            stage.setWidth(width == 0 ? 800 : width);
+            stage.setHeight(height == 0 ? 600 : height);
```

Comment:
I think `width` and `height` is `double` and not `int`. Also, lets not change it. If test wants 0, then pass 0.